apps/lib/publisher_agent/registry.go

```diff
@@ -30,14 +33,31 @@ func (c *RegistryClient) GetBrokersForPublisher(publisherKey signer.PublisherKey
 	if err != nil {
 		return nil, fmt.Errorf("failed to get broker list: %v", err)
 	}
+
 	var brokers []BrokerConnectionConfig
 	err = json.Unmarshal(response, &brokers)
 	if err != nil {
-		return nil, fmt.Errorf("failed to unmarshal broker list: %v", err)
+		// check if the registry returned an error
+		var errorResponse RegistryErrorResponse
+		err = json.Unmarshal(response, &errorResponse)
+		if err == nil {
+			if errorResponse.Error == "Unauthorized" {
+				return nil, fmt.Errorf("not authorized to query Stork Registry - check your configured StorkAuth")
+			} else {
+				return nil, fmt.Errorf("failed to query the Stork Registry: %s", errorResponse.Error)
+			}
+		}
+		return nil, fmt.Errorf("failed to unmarshal response from Stork Registry: %s", string(response))
 	}
 
-	// combine all configs into a single asset map per url
 	brokerMap := make(map[BrokerPublishUrl]map[AssetId]struct{})
+
+	if len(brokers) == 0 {
```

Comment:
make it explicit if the agent is able to reach the registry, but isn't registered